### ISSUE TITLE
Will 2fa work with sessions disabled?

### ISSUE THREAD

**Issue by jwage** [scheb/2fa#190]
**Bundle version**: v6.6.0
**Symfony version**: 5.4.11
**PHP version**: 8.1.18
**Using authenticators** (`enable_authenticator_manager: true`): YES

**Description**

I am disabling built in sessions and csrf protection in my application, so I am changing this configuration:

```
    session:
        storage_factory_id: session.storage.factory.native
        handler_id: Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler
        cookie_secure: auto
        cookie_samesite: lax
        cookie_domain: '%env(COOKIE_DOMAIN)%'
        name: 'TPSESSIONID'
```

and replacing with this:

```
    session: false
```

I am using this with a standard `remember_me` and `form_login`:

```
            remember_me:
                secret:             '%kernel.secret%'
                lifetime:           31536000
                path:               /
                always_remember_me: true
                domain:             '%env(COOKIE_DOMAIN)%'
                name:               'TPREMEMBERME'

            form_login:
                enable_csrf: false
                login_path: login
                check_path: login
                use_referer: true
                default_target_path: login_success
```

When I disable sessions, the 2fa form does not show after I submit the login form. Does scheb/2fa depend on the session currently?

**Comment by scheb** [comment]
2fa depends on the security token being persistent between requests (because it has to store information about the 2fa status of a login somewhere) and that token is stored in the session. So yes, it depends on the session.

**Comment by jwage** [comment]
Any way you can see that I could modify it to store that state in a cookie instead of the session?

**Comment by scheb** [comment]
Hmm, the bundle is not actively managing how the security token is stored. That's happening in Symfony core, i.e. Symfony session management.

Maybe a custom session storage that writes session data straight to a cookie? (Instead of server-side storage and a session id)

**Comment by jwage** [comment]
I will investigate that. Thanks!

**Comment by jwage** [comment]
This is working nicely so far. Thanks for the suggestion!

**Comment by scheb** [comment]
Cool, could you share a bit what you've done? Is there an out of the box solution for this or did you implement it yourself?

**Comment by jwage** [comment]
I basically just implemented a CookieSessionHandler that extends AbstractSessionHandler and instead of writing to pgsql, it writes to a cookie in the response. However, after I posted the comment, I realized I didn't really like the approach because it is exposing stuff from the session data in the cookie and I am not sure if it will be secure or not.

My real problem that led me down this path is the PdoSessionHandler locking. the `SELECT ... FOR UPDATE` queries can sometimes conflict with the garbage collection and cause a deadlock scenario. I don't think I need locking for my sessions since my app doesn't store anything in the session so I am going to try disabling the locking in PdoSessionHandler.
